src-tauri/src/aux_functions/adj_matrix.rs

```diff
@@ -0,0 +1,104 @@
+use crate::graph::graph_ds::Graph;
+use std::collections::HashMap;
+
+/// Converts a graph to an adjacency matrix.
+///
+/// # Arguments
+///
+/// * `graph` - a graph
+///
+/// # Returns
+///
+/// * `Vec<Vec<f64>>` - an adjacency matrix
+pub fn convert_to_adj_matrix(graph: &Graph) -> Vec<Vec<f64>> {
+    let mut adj_matrix: Vec<Vec<f64>> = Vec::new();
+    let nodes = graph.get_nodes();
+    let edges = graph.get_edges();
+
+    let mut ordering_counter = 0 as usize;
+    let mut node_id_to_int = HashMap::new();
+
+    for node in nodes.clone() {
+        node_id_to_int.insert(node.name.clone(), ordering_counter);
+        ordering_counter += 1;
+    }
+
+    for _ in 0..nodes.len() {
+        let mut row: Vec<f64> = Vec::new();
+        for _ in 0..nodes.len() {
+            row.push(0.0);
+        }
+        adj_matrix.push(row);
+    }
+
+    for edge in edges {
+        let u_name = graph.get_node(edge.get_u()).name.clone();
+        let v_name = graph.get_node(edge.get_v()).name.clone();
+
+        let u_id = node_id_to_int.get(&u_name).unwrap();
```

Comment:
Can we use `expect(&str mesg)` instead? In the event this code panics, `unwrap` won't tell the user the human-readable meaning of the problem

Reply:
This would panic if a node is part of an edge but not in the graph. In our add_edge we already give an error if node doesn't exist, so the program would never panic here.